src/zenml/cli/base.py

```diff
@@ -64,18 +68,37 @@ def init(path: Optional[Path]) -> None:
 
 
 @cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
+@click.option("--yes", "-y", is_flag=True, default=False)
 def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+    """Delete all ZenML metadata and artifacts.
+
+    This is a destructive operation, primarily intended for use in development.
 
     Args:
       yes: bool:  (Default value = False)
     """
     if not yes:
-        _ = confirmation(
-            "This will completely delete all pipelines, their associated "
+        confirm = confirmation(
+            "DANGER: This will completely delete all pipelines, their associated "
             "artifacts and metadata ever created in this ZenML repository. "
             "Are you sure you want to proceed?"
         )
 
-    error("Not implemented for this version")
+    if yes or confirm:
+        local_zen_repo_config = Path.cwd() / REPOSITORY_DIRECTORY_NAME
+        global_zen_config = Path(get_global_config_directory())
+        if fileio.exists(str(local_zen_repo_config)):
+            fileio.rmtree(str(local_zen_repo_config))
+            declare(f"Deleted local ZenML config from {local_zen_repo_config}.")
+        if fileio.exists(str(global_zen_config)):
+            config_yaml_path = global_zen_config / "config.yaml"
+            config_yaml_data = yaml_utils.read_yaml(str(config_yaml_path))
+            fileio.rmtree(str(global_zen_config))
+            declare(f"Deleted global ZenML config from {global_zen_config}.")
+
+        fileio.makedirs(str(global_zen_config))
+        yaml_utils.write_yaml(str(config_yaml_path), config_yaml_data)
+        Repository.initialize(root=Path.cwd())
+
+    else:
+        declare("Aborting clean.")
```

Comment:
I would generally also log.info exactly what is deleted (i.e. each folder). To be more verbose about the deletion.

Reply:
Done.